base/range.jl

```diff
@@ -451,14 +451,21 @@ if isdefined(Main, :Base)
     end
 end
 
+"""
+    Base.AbstractOneTo
+
+Abstract type for ranges that start at 1.
+"""
+abstract type AbstractOneTo{T} <: AbstractUnitRange{T} end
```

Comment:
Should there be an `<:Integer` upper bound on the parameter?

```suggestion
abstract type AbstractOneTo{T<:Integer} <: AbstractUnitRange{T} end
```

Reply:
I'd say it's better to leave it unrestricted. This may allow, e.g., using `StaticInt` types from `Static.jl` that don't subtype `Integer`.

Reply:
`StaticInt` doesn't subtype `Integer` merely as a workaround for invalidation in `Base` that would cause. That was my takeaway from reading the old discussions on Github. So, IMO:
* `StaticInt` not subtyping `Integer` was a hacky quick fix in the first place
* long-term `StaticInt` should subtype `Integer`

Reply:
Is there any particular advantage to subtyping `Integer` here in the abstract type?

Reply:
I agree there should be no upper bound on the type parameter, for consistency with other abstract range types.